### DISCUSSION TITLE
Do not edit the Races_Pokemon.xml file

### DISCUSSION THREAD

**Discussion by Gargamiel** [Gargamiel/PokeWorld#15] [Announcements]
You are probably wondering how I added all 493 Pokémon into RimWorld and more specifically how I wrote the 78546 lines of the Races_Pokemon.xml file.

I of course didn't write all of this by hand for a year while making this mod. I basically wrote a python script that fetch a whole bunch of data from an excell table containing Pokémon data and write / populates all the defs with these data. This allows me to quickly edit any value within the def file in a single click and in half a second. 

Lets say for example I want to slightly increase the move speed in RimWorld of all Pokémon. I just have to tweak a little bit my algorithm computing the speed (a simple formula that turns Pokémon base speed into RimWorld speed) and I easily buff / nerf the speed of every Pokémon at once.

You probably understand where I'm going with this: if a contributor makes a pull request editing the race file, I'll overwrite any modification done to it if I use my script to change something after that.

Other xml files containing defs also have the same issue, but these are smaller: the eggs file PokemonEggs.xml, the Pokémon sound files, and the spawn rate patch Patch_BiomeWildAnimals.xml

This wasn't an issue while I was working alone, but now that the mod is gaining traction and contributors, it pauses a real issue that I am really willing to fix. I am just going to need a bit of time to figure the right way to do so. I'm not sure if I want to directly add the python scripts and excel database to this repo, create another repo for those files, or what else to do.

If you have any idea of how to manage this transition to a better work environment for all of us, please share your ideas. Once we'll have resolved this issue, we'll be able to go full speed to add more content, more Pokémon, and more fun for every users of this awesome mod!

ps: my excell database and python scripts are absolutely disgusting, so I'll probably have to refactor them a little before sharing them.

**Comment by lmgginspace** [comment]
That's understandable given the amount of Pokémon there are ^^
That seems like a good idea IMHO, to clean a bit both the script and the database and then make a separate repo. And of course, as with anything related to programming, not rushing things.

**Reply by Gargamiel** [reply]
Heads up! I just released most of the scripts and data I used to generate the defs, you can find them here: https://github.com/Gargamiel/PokeWorld_Scripts

The biggest script generating the thingdefs is already there, it's just missing the scripts used to generate the sounddef and the biome spawn rate for each Pokémon.
